qemu/tests/thin_provisioning_optimize.py

```diff
@@ -0,0 +1,59 @@
+import logging
+import os
+import re
+
+
+from avocado.utils import process
+from avocado.core import exceptions
+
+from autotest.client.shared import error
+
+from virttest import data_dir
+from virttest import utils_misc
+
+
+@error.context_aware
+def run(test, params, env):
+
+    """
+      'thin-provisioning' functions test using sg_utils:
+      1) Create image using qemu-img
+      2) Convert the image and check if the speed is much
+         faster than standard time
+
+      :param test: QEMU test object
+      :param params: Dictionary with the test parameters
+      :param env: Dictionary with test environment.
+      """
+
+    standard_time = 0.4
+    qemu_img_binary = utils_misc.get_qemu_img_binary(params)
+    base_dir = params.get("images_base_dir", data_dir.get_data_dir())
+    if not qemu_img_binary:
+        raise exceptions.TestError("Can't find the command qemu-img.")
+
+    image_create_cmd = params["create_cmd"]
+    image_create_cmd = image_create_cmd % (qemu_img_binary, base_dir)
+    image_convert_cmd = params["convert_cmd"]
+    image_convert_cmd = image_convert_cmd % (
+        qemu_img_binary, base_dir, base_dir)
+    process.system(image_create_cmd, shell=True)
+    output = process.system_output(image_convert_cmd, shell=True)
+    realtime = re.search(r"real\s+\dm(.*)s", output)
+    if realtime is None:
+        raise exceptions.TestError(
+            "Faild to get the realtime from {}".format(output))
+    realtime = float(realtime.group(1))
+    logging.info("real time is : {:f}".format(realtime))
+    if realtime >= standard_time:
```

Comment:
where you get stand time?  image create time depends on disk where your image keep or image_size and image_format and so on...  so, how to ensure your test is right.